src/lib/common/NotificationFormat.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+*
+* Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <string.h>
+#include <string>
+#include <sstream>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+#include "common/NotificationFormat.h"
+#include "common/wsStrip.h"
+
+
+
+/* ****************************************************************************
+*
+* notificationFormatToString - 
+*/
+const char* notificationFormatToString(NotificationFormat format, bool noDefault)
```

Comment:
Pre-defining the string that the function may return add complexity, so in this case it seems better to use `const  char*` at the end.

What I wonder if we should consolidate a cosing style rule based in this experience (for not passing again the same discussion in a couple of PRs), maybe this:

> However, that would go against our 'general' rule to use const std::string& for input strings and char* for output strings ...